tx-generator/src/main.rs

```diff
@@ -0,0 +1,123 @@
+use mixed_workload::{MIXED_DISTRIBUTION, WRITE_LENGTH_DISTRIBUTION};
+use nft_workload::{NFT_CONTRACT_DISTRIBUTION, NFT_USER_DISTRIBUTION};
+use rand::{
+    distr::{weighted::WeightedIndex, Distribution},
+    rng,
+    seq::IndexedRandom,
+    Rng,
+};
+use uniswap_workload::{AVERAGE_VALUE_DISTRIBUTION, BURSTY_VALUE_DISTRIBUTION};
+
+use crate::p2p_workload::{RECEIVER_DISTRIBUTION, SENDER_DISTRIBUTION};
+
+mod mixed_workload;
+mod nft_workload;
+mod p2p_workload;
+mod uniswap_workload;
+
+/// Returns the percentile of a sorted vector.
+fn percentile(sorted_vec: &[f64], percentile: f64) -> Option<f64> {
+    let len = sorted_vec.len();
+    if len == 0 {
+        return None;
+    }
+
+    let index = ((percentile / 100.0) * len as f64).ceil() as usize - 1;
+    Some(sorted_vec[index.min(len - 1)])
+}
+
+/// Solana workload.
+/// Generate a transaction with a random number of inputs (ignoring reads).
+fn solana_concurrency<R: Rng>(rng: &mut R) {
```

Comment:
The execution complexity is missing here (transactions with longer runtime have a higher chance to overlap)

Reply:
what do you mean by execution complexity? How can I find it?

Reply:
https://github.com/ISTA-SPiDerS/chain-execution-benchmark/blob/version-1.0/workloads/mixed/mixed_workload.rs#L5

The Gas Cost distribution. Functions the same as the write length distribution, but is about the execution complexity in gas.

(Interestingly, the execution complexity is decoupled from the number of writes)

Reply:
ho I see. Do we roughly have a mapping between a unit of gas spent vs average time spent to run the computation?

Reply:
No, unfortunately I wasn't able to find that, and as it's also difficult to apply that across different VMs, I went with a "relative" measure. (i.e. for each unit of gas a read/write loop).

Reply:
done, updated